File: frontend/src/app/services/task.service.ts
import { Injectable } from '@angular/core';
import { Task } from '../models/task.model';

@Injectable({
  providedIn: 'root',
})
export class TaskService {
  private apiUrl = 'http://localhost:8080/api/tasks';

  async getTasks(): Promise<Task[]> {
    const res = await fetch(this.apiUrl);
    if (!res.ok) throw new Error('Failed to load tasks');
    return res.json();
  }

  async createTask(title: string, description: string): Promise<Task> {
    const res = await fetch(this.apiUrl, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ title, description }),
    });
    if (!res.ok) throw new Error('Failed to create task');
    return res.json();
  }

  async moveTask(id: number, status: 'TODO' | 'IN_PROGRESS' | 'DONE'): Promise<Task> {
    const res = await fetch(`${this.apiUrl}/${id}/move/${status}`, {
      method: 'PUT',
    });
    if (!res.ok) throw new Error('Failed to move task');
    return res.json();
  }
}

File: backend/src/main/java/com/taskpulse/service/TaskService.java
package com.taskpulse.service;

import com.taskpulse.model.Task;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.atomic.AtomicLong;

public class TaskService {

    private static final List<Task> tasks = new ArrayList<>();
    private static final AtomicLong COUNTER = new AtomicLong(1);

    public List<Task> getAll() {
        return tasks;
    }

    public Task create(String title, String description) {
        Task task = new Task(
                COUNTER.getAndIncrement(),
                title,
                description,
                "TODO"
        );
        tasks.add(task);
        return task;
    }

    public Task move(Long id, String status) {
        for (Task task : tasks) {
            if (task.id.equals(id)) {
                task.status = status;
                return task;
            }
        }
        return null;
    }
}
